### ISSUE TITLE
언어별 자주사용하는 라이브러리 및 함수 정리

### ISSUE THREAD

**Issue by acious** [GDGKoreaCampus/study_algo#2]
알고리즘 풀이만을 위해서 각 언어별로 자주 사용하는 라이브러리 및 함수를 정리하면 어떨까요?

예를들면 Python에서 상황별로 데이터를 입,출력하는 것부터 시작해서 자료구조를 바로 가져와서 쓸수있는 라이브러리를 정리하면 좋지 않을까하네요.

물론 Wiki에 정리하는게 제일 좋아보입니다!

**Comment by gnujoow** [comment]
:) 의견 감사합니다. 좋은 의견인것 같습니다.

**Comment by gnujoow** [comment]
@acious 작성은 어떤방식으로 하는것이 좋을까요? 자료구조의 구현을 이 언어에서는 이런식으로 한다식으로 정리하면 편할까요?

이를테면

> - 트리
>   - **C/C++** 행렬, [리스트](http://www.cplusplus.com/reference/list/list/)
>   - **python** rbtree

**Comment by acious** [comment]
자주쓰이는 각 자료구조나 도구마다 note를 만들고 해당 노트안에 언어별로 작성하면되지 않을까 싶네요 ㅎㅎ

ex)
- 입출력
  - C++
  - Python
  - Go
- 리스트
  - C++
  - Python
  - Go

**Comment by acious** [comment]
test

**Comment by mingrammer** [comment]
그럼 '입출력 - Python'과 같은 문서들이 생성되는건가요? 이들을 하나의 문서에서 링크로 연결하고

**Comment by acious** [comment]
문서는 입출력이라는 문서하나만 존재하고, 그 안에 컨텐츠로 각 언어별로 마크다운을 이용하여 짧게 쓰면되지 않을까요 ㅎㅎ? 물론 길어진다면... 문제가 될까요?

**Comment by mingrammer** [comment]
네 그게 나을 것 같습니다. 다 따로 나누면 너무 파편화가 되어버리니 .. 소주제로만 나누고 안에서 여러 언어로 작성하는게 좋을 것 같습니다. (한 눈에 보기도 쉽구요.)

**Comment by gnujoow** [comment]
https://github.com/GDGKoreaCampus/study_algo/issues/2#issuecomment-251867222 @mingrammer 좋은 의견인것 같습니다.

작성되는 내용은 어떤게 있으면 좋을까요?
제 생각에는 입출력과 기본적인 자료구조의 구현(이진트리, 그래프, 스택, 큐)와 표준라이브러리에서 제공하는 알고리즘들 (quick sort, merge sort)을 작성하면 좋을것 같은데 다른분들 생각은 어떠신지 여쭙고 싶습니다.

**Comment by acious** [comment]
네 ㅎㅎ 일단 기본적인것부터 시작하는게 우선일것같네요.  
단 import로 간단히 해결가능한것은 그렇게 가져다 쓰면 된다고 안내만 해주면 되지 않을까합니당.

**Comment by mingrammer** [comment]
그 정도면 될 것 같습니다. 기본적인 자료구조나 알고리즘은 위키피디아처럼
- Binary Tree
  - Python
  - C++
  - Go
    ...

로 구현체들을 보여주면서, 표준 라이브러리에 해당 구현체가 있을경우 (가능한 경우) 링크와 함께 알려주면 되지않을까 합니다.

**Comment by mingrammer** [comment]
Dequeue의 구현체를 보여주면서 Python의 queue라이브러리를 링크와 함께 언급하는등 ..

**Comment by acious** [comment]
C++, Java, Python부터 시작하면 되지 않을까요?

**Comment by mingrammer** [comment]
자기가 잘하거나 잘아는 언어가 있으면 뭐 바로 추가해도 될 것 같습니다. Go언어 잘하시면 Go도 바로 추가할 수 있고 .. 등등.. 저는 Java를 모르기에 Java는 잘 아시는분이 해주시면 될 것 같구요 ㅎㅎ

**Comment by acious** [comment]
Java랑 Go를 제가 맡겠습니다!
